pkg/apis/clusteroperator/v1alpha1/types.go

```diff
@@ -47,109 +48,278 @@ type ClusterList struct {
+	MachineSets []ClusterMachineSet `json:"machineSets"`
+}
+
+// ClusterHardwareSpec specifies hardware for a cluster. The specification will
+// be specific to each cloud provider.
+type ClusterHardwareSpec struct {
+	// AWSConfig specifies cluster hardware configuration on AWS
 	// +optional
-	ComputeNodeGroups []ClusterComputeNodeGroup `json:"computeNodeGroups,omitempty"`
+	AWS *AWSClusterSpec `json:"aws"`
+
+	// TODO: Add other cloud-specific Specs as needed
 }
 
-type ClusterStatus struct {
-	// MasterNodeGroups is the number of actual master node groups that are
-	// active for the cluster
-	MasterNodeGroups int `json:"masterNodeGroups"`
+// AWSClusterSpec contains cluster-wide configuration for a cluster on AWS
+type AWSClusterSpec struct {
+	// AccountSeceret refers to a secret that contains the AWS account access
+	// credentials
+	AccountSecret corev1.LocalObjectReference `json:"accountSecret"`
+
+	// Region specifies the AWS region where the cluster will be created
+	Region string `json:"region"`
+
+	// VPCName specifies the name of the VPC to associate with the cluster.
+	// If existing, it will not be created. If left blank, the VPC name will
+	// be derived from the cluster name.
+	// +optional
+	VPCName string `json:"vpcName"`
 
-	// ComputeNodeGroups is the number of actual compute node groups that are
-	// active for the cluster
-	ComputeNodeGroups int `json:"computeNodeGroups"`
+	// VPCSubnet specifies the subnet to use for the cluster's VPC. Only used
+	// when a new VPC is created for the cluster
+	VPCSubnet string `json:"vpcSubnet"`
 }
 
-// ClusterNodeGroup is a node group defined in a Cluster resource
-type ClusterNodeGroup struct {
-	Size int `json:"size"`
+// ClusterConfigSpec contains OpenShift configuration for a cluster
+type ClusterConfigSpec struct {
+	// DeploymentType indicates the type of OpenShift deployment to create
+	DeploymentType string `json:"deploymentType"`
+
+	// OpenShiftVersion is the version of OpenShift to install
+	OpenshiftVersion string `json:"openshiftVersion"`
+
+	// SDNPluginName is the name of the SDN plugin to use for this install
+	SDNPluginName string `json:"sdnPluginName"`
+
+	// ServiceNetworkSubnet is the CIDR to use for service IPs in the cluster
+	ServiceNetworkSubnet string `json:"serviceNetowrkSubnet"`
+
+	// PodNetworkSubnet is the CIDR to use for pod IPs in the cluster
+	PodNetworkSubnet string `json:"podNetworkSubnet"`
 }
 
-// ClusterComputeNodeGroup is a compute node group defined in a Cluster
-// resource
-type ClusterComputeNodeGroup struct {
-	ClusterNodeGroup `json:",inline"`
+// ClusterStatus contains the status for a cluster
+type ClusterStatus struct {
+	// MachineSetCount is the number of actual machine sets that are active for the cluster
+	MachineSetCount int `json:"machineSetCount"`
 
-	Name string `json:"name"`
+	// MasterMachineSetName is the name of the master machine set
+	MasterMachineSetName string `json:"masterMachineSetName,omitempty"`
```

Comment:
The latter. They are set by the ClusterController when syncing a Cluster. They contain the names of the actual MachineSets that are designated as master and infra. My thinking with this was that it would be useful when generating inventories. They would also help in the case of the MachineSetController checking whether for a given cluster the master MachineSet is ready and it can proceed to provision compute machineSets. Sure, we can find this by querying MachineSets owned by a cluster, but this is information we have available to us when calculating the Status of a cluster, so why not use it.

Reply:
Ah, I see where I got confused. This is the full, realized name of the machine set. It is not the name from `ClusterMachineSet`. OK. It makes more sense to me to have it then. I still am not entirely convinced that they are worthwhile, though, based on my understanding of how things will work. In both of the cases that you provided, the controller wouldn't make use of the name. Inventory and ready checking will both be done using tags applied to the auto scaling group. They won't be doing a query based upon the kube name of the machine set. In any event, I am fine with leaving them in for now.